lwgenerate/azure/azure.go

```diff
@@ -0,0 +1,460 @@
+package azure
+
+import (
+	"github.com/hashicorp/hcl/v2/hclwrite"
+	"github.com/lacework/go-sdk/lwgenerate"
+	"github.com/pkg/errors"
+)
+
+type GenerateAzureTfConfigurationArgs struct {
+	// Should we configure Activity Log integration in LW?
+	ActivityLog bool
+
+	// Should we add Config integration in LW?
+	Config bool
+
+	// Should we create an Active Directory integration
+	AdIntegration bool
+
+	// If Config is true, give the user the opportunity to name their integration. Defaults to "TF Config"
+	ConfigIntegrationName string
+
+	// If ActivityLog is true, give the user the opportunity to name their integration. Defaults to "TF activity log"
+	ActivityLogIntegrationName string
+
+	// Active Directory application Id
+	AdApplicationId string
+
+	// Active Directory password
+	AdApplicationPassword string
+
+	// Active Directory Enterprise app object id
+	AdServicePrincipalId string
+
+	// Should we use the management group, rather than subscription
+	ManagementGroup bool
+
+	// Management Group ID to set
+	ManagementGroupId string
+
+	// List of subscription Ids
+	SubscriptionIds []string
+
+	// Grant read access to ALL subscriptions
+	AllSubscriptions bool
+
+	// Storage Account name
+	StorageAccountName string
+
+	// Storage Account Resource Group
+	StorageAccountResourceGroup string
+
+	// Should we use existing storage account
+	ExistingStorageAccount bool
```

Comment:
For these settings, can we just use the presence of a value instead of having these `bool` values as well?

i.e., if `ManagementGroupId` is not `""`; then we should use it.  Or for `AllSubscriptions`, if `SubscriptionIds` is empty then all subs get set.  

If not, all of that logic is going to have to be handled further 'downstream' in the CLI code when we survey the end-user.

Reply:
Yes, certainly in the case of the `ManagementGroup` and `ExistingStorageAccount` flags, then the presence of the elements set will indicate if the the flag was set or not. The `AllSubscriptions` flag is a little different as it is set in the hcl for the config and activity log to indicate that all subscriptions within the selected tenant have read access, if this is not set it will only use the primary subscription (unless a list of subscriptions is given)

Reply:
I'm certainly happy with leaving it as-is if that's beneficial to your other existing code.  The only reason I raised it is that these `bool` properties had caused myself and @rmoles some challenges previously (in the interactive CLI code).